### PR TITLE
x11-wm/evilwm: Bump to EAPI=6

### PR DESCRIPTION
**Pull request by charIes17**

### CODE REVIEW

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	sed -i -e '/^CFLAGS/s/ -Os/ /' \
+		-e 's/install -s /install /' Makefile || die "sed opt failed"
+}
+
+src_compile() {
+	emake CC="$(tc-getCC)" prefix="\$(DESTDIR)/${EPREFIX}/usr" XROOT="${EPREFIX}/usr" LDPATH="-L${EPREFIX}/usr/$(get_libdir)" || die
+}
+
+src_install () {
+	make DESTDIR="${D}" prefix="\$(DESTDIR)/${EPREFIX}/usr" install || die "make install failed"
```
Comment (monsieurp) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: Please call `emake` instead.

Comment (monsieurp) [reply x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: And get rid of `die` at the end.

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+	dodoc ChangeLog README TODO || die "dodoc failed"
```
Comment (monsieurp) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: Remove this line and call `einstalldocs` instead.

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+	echo -e "#!${EPREFIX}/bin/sh\nexec \"${EPREFIX}/usr/bin/${PN}\"" > "${T}/${PN}"
```
Comment (monsieurp) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: Please call `|| die` after the command.

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	exeinto /etc/X11/Sessions
+	doexe "${T}/${PN}" || die "/etc/X11/Sessions failed"
```
Comment (monsieurp) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: Remove `die` here please.

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+	insinto /usr/share/xsessions
+	doins "${FILESDIR}/${PN}.desktop" || die "${PN}.desktop failed."
```
Comment (monsieurp) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: `die` can be removed here as well.

Comment (charIes17) [reply x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: All changes done as requested

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+src_prepare() {
+	default
+	sed -i 's/^#define DEF_FONT.*/#define DEF_FONT "fixed"/' evilwm.h \
+		|| die "sed font failed"
+	sed -i -e '/^CFLAGS/s/ -Os/ /' \
+		-e 's/install -s /install /' Makefile || die "sed opt failed"
+}
+
+src_compile() {
+	emake CC="$(tc-getCC)" prefix="\$(DESTDIR)/${EPREFIX}/usr" XROOT="${EPREFIX}/usr" LDPATH="-L${EPREFIX}/usr/$(get_libdir)" || die
```
Comment (jonasstein) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: emake needs no `die` too. See https://devmanual.gentoo.org/ebuild-writing/eapi/

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	emake DESTDIR="${D}" prefix="\$(DESTDIR)/${EPREFIX}/usr" install
+
+	einstalldocs
+
+	echo -e "#!${EPREFIX}/bin/sh\nexec \"${EPREFIX}/usr/bin/${PN}\"" > "${T}/${PN}" || die
+	exeinto /etc/X11/Sessions
+	doexe "${T}/${PN}"
+
+	insinto /usr/share/xsessions
+	doins "${FILESDIR}/${PN}.desktop"
```
Comment (jonasstein) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: don't you want a domenu here? https://devmanual.gentoo.org/eclass-reference/eutils.eclass/

**Comment by monsieurp** [comment]
@charIes17 ping.

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild (hunk 7 above)
Comment (charIes17) [reply x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: Finally, domenu added from filesdir

File: x11-wm/evilwm/evilwm-1.0.1-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit eutils toolchain-funcs multilib
```
Comment (Amynka) [inline x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: Do you really need multilib eclass?

Comment (SoapGentoo) [reply x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: also switch to `desktop.eclass` instead of `eutils`

Comment (charIes17) [reply x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: eutils replaced with desktop

Comment (charIes17) [reply x11-wm/evilwm/evilwm-1.0.1-r1.ebuild]: And multilib removed

**Pull request closed without merging**